analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -61,9 +62,106 @@ protected override ProgramState PreProcess(ProgramState state, IBinaryOperationW
     {
         BinaryOperatorKind.Add => NumberConstraint.From(left.Min + right.Min, left.Max + right.Max),
         BinaryOperatorKind.Subtract => NumberConstraint.From(left.Min - right.Max, left.Max - right.Min),
+        BinaryOperatorKind.Multiply => CalculateMultiply(left, right),
+        BinaryOperatorKind.And when left.IsSingleValue && right.IsSingleValue => NumberConstraint.From(left.Min.Value & right.Min.Value),
+        BinaryOperatorKind.And => CalculateAnd(left, right),
```

Comment:
This is getting big. We'll need to split this class. Let's move the `Calculate` and everything that it invokes into a partial class and `Binary.Math.cs` or `Binary.Arithmetic.cs` or something like that.

With the compound assignment operators, it might get extracted elsewhere afterwards too.